pkg/nmi/server/server.go

```diff
@@ -244,7 +271,7 @@ func (s *Server) getMICToken(logger *log.Entry, rqClientID, rqResource string) (
 		// TODO: return the right status code based on the error we got from adal.
 		return nil, http.StatusForbidden, err
 	}
-	response, err := json.Marshal(*token)
+	response, err := json.Marshal(newMSIResponse(*token))
```

Comment:
@lodejard Thank you for this PR. We can get the sdk version from the request url. The key name is `api-version`. I think we should make a decision based on the api-version and return the appropriate token response.

Reply:
You can use this - https://github.com/Azure/aad-pod-identity/blob/master/pkg/nmi/server/server.go#L369-L376 to also get the `api-version` and then make a decision if we need to send the token as is or create a new MSI response.

Reply:
(merging thread with above)